gensim/matutils.py

```diff
@@ -895,8 +895,11 @@ def docbyoffset(self, offset):
             assert previd <= docid, "matrix columns must come in ascending order"
             if docid != previd:
                 if previd >= 0:
-                    return document
+                    break
```

Comment:
Can you add a test for this?

Reply:
Hmm, ok. But I don't understand what exactly should I check in it ?

Reply:
I read full method definition, now it's clear for me, thanks.